src/esp/sensor/SensorFactory.cpp

```diff
@@ -0,0 +1,18 @@
+#include "esp/sensor/SensorFactory.h"
+
+#include "esp/scene/SceneNode.h"
+#include "esp/sensor/CameraSensor.h"
+#include "esp/sensor/Sensor.h"
+
+namespace esp {
+sensor::SensorSuite SensorFactory::createSensors(
+    scene::SceneNode& node,
+    const sensor::SensorSetup& sensorSetup) {
+  sensor::SensorSuite sensorSuite = sensor::SensorSuite();
+  for (sensor::SensorSpec::ptr spec : sensorSetup) {
```

Comment:
This is already decoupled on the Python side FYI. It's even factored out on the Python side to a sensor can be added without reconfiguring the sim. Seems like this should be refactored to also have a function to add a single sensor like we have on the Python side ```agent.add_sensor()...```

Reply:
Hi @Skylion007 
If possible, I sincerely hope you can help us to take a look at the internal task (I will send you the task number via slack), where we gave concrete implementation steps. Check if something is contradictory to what has been done in the Python side. Thanks!

Reply:
> This is already decoupled on the Python side FYI. It's even factored out on the Python side to a sensor can be added without reconfiguring the sim. Seems like this should be refactored to also have a function to add a single sensor like we have on the Python side `agent.add_sensor()...`

As an aside, it looks like the sensor creation for the agent in agent.py happens in agent.reconfigure when the agent is created, regardless of whether created sensors are passed into the class or not, which seems like a bug, no? 

Pertinent Agent.py agent code

```
    def __init__(
        self,
        scene_node: hsim.SceneNode,
        agent_config: Optional[AgentConfiguration] = None,
        _sensors: Optional[SensorSuite] = None,
        controls: Optional[ObjectControls] = None,
    ) -> None:
        self.agent_config = agent_config if agent_config else AgentConfiguration()
        
        # Assigning pre-existing sensors is ignored in below reconfigure call
        self._sensors = _sensors if _sensors else SensorSuite()
< ... >        
        # Reconfigure always called with default reconfigure_sensors == True
        self.reconfigure(self.agent_config)
< ... >        

    def reconfigure(
        self, agent_config: AgentConfiguration, reconfigure_sensors: bool = True
    ) -> None:
< ... >        
        # This is always True if called from self.reconfigure called from constructor
        if reconfigure_sensors:
            self._sensors.clear()
            for spec in self.agent_config.sensor_specifications:
                self._add_sensor(spec, modify_agent_config=False)

```